### PR TITLE
update inteerface-go-ipfs-core and handle breaking changes

### PR DESCRIPTION
**Pull request by Stebalien**
* No more Hidden, StdinName, and Wrap options.
* LsLink -> DirEntry with file types that don't expose internals.

### CODE REVIEW

File: unixfs.go [hunk 1]
```diff
@@ -62,13 +60,8 @@ func (api *UnixfsAPI) Add(ctx context.Context, f files.Node, opts ...caopts.Unix
...
 
-	switch c := f.(type) {
-	case files.Directory:
-		req.Body(files.NewMultiFileReader(c, false))
-	case files.File:
-		d := files.NewMapDirectory(map[string]files.Node{"": c}) // unwrapped on the other side
-		req.Body(files.NewMultiFileReader(d, false))
-	}
+	d := files.NewMapDirectory(map[string]files.Node{"": f}) // unwrapped on the other side
+	req.Body(files.NewMultiFileReader(d, false))
```
Comment (Stebalien) [inline unixfs.go]: Did we break something here? I had to do this to get the API tests to pass.

Comment (Stebalien) [reply unixfs.go]: https://github.com/ipfs/interface-go-ipfs-core/pull/25

Comment (Stebalien) [reply unixfs.go]: cc @magik6k?

**Comment by aknuds1** [comment]
Thanks for this PR! By applying it in my fork, I could finally upgrade our project to latest go-ipfs, which is necessary for us to switch to Go modules.

File: unixfs.go (hunk 1 above)
Comment (Stebalien) [reply unixfs.go]: The way I see it, we're giving ownership of the file to the add API.

**Pull request merged**